plugins/animate/animate.cpp

```diff
@@ -243,13 +286,78 @@ class wayfire_animation : public wf::singleton_plugin_t<animation_global_cleanup
         return {"none", 0};
     }
 
+    bool try_reverse(wayfire_view view, wf_animation_type type, std::string name,
+        int direction)
+    {
+        direction = !direction;
+        if (view->has_data(name))
+        {
+            auto data = view->get_data<animation_hook_base>(name);
+            if (data->get_direction() == direction)
+            {
+                data->reverse(type);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+#define FORWARD 1
```

Comment:
1) please don't use macros when constexpr would suffice
2) nitpicking but: forward seems less clear than 'SHOWN', after all, the forward state is that the view is SHOWN